cdap-watchdog/src/main/java/co/cask/cdap/logging/read/FileLogReader.java

```diff
@@ -88,24 +73,18 @@ public void getLogNext(final LoggingContext loggingContext, final ReadRange read
       long fromTimeMs = readRange.getFromMillis() + 1;
 
       LOG.trace("Using fromTimeMs={}, readRange={}", fromTimeMs, readRange);
-      NavigableMap<Long, Location> sortedFiles = fileMetaDataManager.listFiles(loggingContext);
-      if (sortedFiles.isEmpty()) {
+      List<LogLocation> sortedFilesInRange =
+        fileMetadataReader.listFiles(getLogPathIdentifier(loggingContext),
+                                     readRange.getFromMillis(), readRange.getToMillis());
+      if (sortedFilesInRange.isEmpty()) {
         return;
       }
 
-      List<Location> filesInRange = getFilesInRange(sortedFiles, readRange.getFromMillis(), readRange.getToMillis());
-      AvroFileReader logReader = new AvroFileReader(schema);
```

Comment:
Why dont we use AvroFileReader rather than copy pasting the same code in `LogLocation` ?